tensorflow_addons/utils/tensor_util.py

```diff
@@ -0,0 +1,319 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
```

Comment:
do we really need to copy over tensor_util to tf addons instead of using the one in TF core?

Reply:
The only usage of it was check shape, do we have any tf core utility to do that? I would love to use any code in core.

Reply:
is it safe for tf-addons to use nonpublic API bits?

Reply:
We probably should move away from nonpublic API since it can be changed anytime and not protected by the API contract.